mapper-processor/src/main/java/com/datastax/oss/driver/internal/mapper/processor/DecoratedMessager.java

```diff
@@ -37,135 +39,93 @@ public void warn(String template, Object... arguments) {
     messager.printMessage(Diagnostic.Kind.WARNING, String.format(template, arguments));
   }
 
-  /** Emits a warning for a type. */
-  public void warn(TypeElement typeElement, String template, Object... arguments) {
-    messager.printMessage(Diagnostic.Kind.WARNING, String.format(template, arguments), typeElement);
+  /** Emits a warning for a specific element. */
+  public void warn(Element element, String template, Object... arguments) {
+    message(Diagnostic.Kind.WARNING, element, template, arguments);
   }
 
-  /** Emits an error for a type. */
-  public void error(TypeElement typeElement, String template, Object... arguments) {
-    messager.printMessage(Diagnostic.Kind.ERROR, String.format(template, arguments), typeElement);
+  /** Emits an error for a specific element. */
+  public void error(Element element, String template, Object... arguments) {
+    message(Diagnostic.Kind.ERROR, element, template, arguments);
   }
 
-  /**
-   * Emits a warning for a program element that might be inherited from another type.
-   *
-   * @param targetElement the element to target.
-   * @param processedType the type that we were processing when we detected the issue.
-   */
+  /** @deprecated TODO inline this once JAVA-2721 and JAVA-2816 are merged */
+  @Deprecated
   public void warn(
       Element targetElement, TypeElement processedType, String template, Object... arguments) {
```

Comment:
I've left these two methods to avoid unrelated impacts for now, but they should be inlined after this is merged together with the other pending mapper PRs.